torch/_dynamo/variables/tensor.py

```diff
@@ -996,6 +999,15 @@ def python_type(self):
     def as_proxy(self):
         return self.proxy
 
+    def as_tensor(self, tx):
+        if self._tensor_var is None:
+            from .builder import SourcelessBuilder
+
+            self._tensor_var = SourcelessBuilder.create(
+                tx, torch.scalar_tensor
+            ).call_function(tx, [self], {})
```

Comment:
Should we pass here `device="cpu"`? My concern is that, what if this is being run in a context where a global device is set, like we recommend in the case users want to execute their NumPy code on GPU?

OTOH, I'm not completely sold on the idea, as there may be other devices that don't play ball with CPU scalars?

Reply:
Can you remind me again how the Numpy code on GPU stuff works

Reply:
See https://pytorch.org/docs/stable/torch.compiler_faq.html#can-i-execute-numpy-code-on-cuda-and-compute-gradients-via-torch-compile

In particular, see the code:
```python
@torch.compile
@torch.compiler.wrap_numpy
def numpy_fn(X, Y):
    prod = X[:, :, None] * Y[:, None, :]
    print("oops, a graph break!")
    return np.sum(prod, axis=(-2, -1))

X = torch.randn(1024, 64, device="cuda")
Y = torch.randn(1024, 64, device="cuda")

with torch.device("cuda"):
    Z = numpy_fn(X, Y)
assert isinstance(Z, torch.Tensor)
assert Z.device.type == "cuda"
```